pkg/ingester/client/chunkcompat.go

```diff
@@ -65,6 +66,29 @@ func StreamsToMatrix(from, through model.Time, responses []*QueryStreamResponse)
 	return result, nil
 }
 
+// StreamingSeriesToMatrix converts slice of []client.TimeSeriesChunk to a model.Matrix.
+func StreamingSeriesToMatrix(from, through model.Time, sSeries []StreamingSeries) (model.Matrix, error) {
+	if sSeries == nil {
+		return nil, nil
+	}
+
+	result := model.Matrix{}
+	for _, series := range sSeries {
+		for sourceIdx, source := range series.Sources {
+			chunks, err := source.StreamReader.GetChunks(source.SeriesIndex)
+			if err != nil {
+				return nil, fmt.Errorf("GetChunks() from stream reader for series %d from source %d: %w", source.SeriesIndex, sourceIdx, err)
+			}
+			stream, err := seriesChunksToMatrix(from, through, series.Labels, chunks)
+			if err != nil {
+				return nil, err
+			}
+			result = append(result, stream)
+		}
+	}
```

Comment:
This isn't quite right - we'll add a series per source ingester, rather than one series with chunks from all sources. 

We need to accumulate all chunks for a series from all sources, then call `seriesChunksToMatrix` on them.

Reply:
That was the missing piece :)
Thank you @charleskorn. The test finally passes.